thirdeye-server/src/main/java/ai/startree/thirdeye/service/DataSourceService.java

```diff
@@ -175,4 +177,8 @@ public boolean validate(final ThirdEyePrincipal principal, final long id) {
     authorizationManager.ensureCanRead(principal, dataSourceDto);
     return dataSourceCache.getDataSource(dataSourceDto).validate();
   }
+
+  public DataSourceApi recommend(final ThirdEyePrincipal principal) {
+    return authorizationManager.generateDatasourceConnection(principal);
```

Comment:
let's have an `authorizationManager.ensureCanRead` call first to ensure the user can read datasources. Else it could leak Pinot tokens.

as of today the design is not that great, so you will have to create a dummy datasource object :/,  see https://github.com/startreedata/thirdeye/blob/d8affb568fd3e8c7977be4208a9be6151591197b/thirdeye-server/src/main/java/ai/startree/thirdeye/service/AnomalyService.java#L100

Reply:
Yeah, i was thinking to add something like this
but then thought we there will be checks at time of creation
but yeah we need to protect from pinot token data as well